File: code/java/ClassConstructor.java
class Animal {
  private String name;

  public Animal(String name) {
    this.name = name;
  }

  public void walk() {
    System.out.println("My name is " + this.name + " and I'm walking");
  }
}

class ClassConstructor {
  public static void main(String[] args) {
    new Animal("Kelya").walk();
  }
}

File: code/php/ClassConstructor.php
<?php

class AnimalB
{
    private $name;

    public function __construct(string $name)
    {
        $this->name = $name;
    }

    public function walk()
    {
        echo "My name is {$this->name} and I'm walking";
    }
}

$animal = new AnimalB("Kelya");
$animal->walk();